src/main/java/com/acrolinx/sidebar/lookup/LookupForResolvedEditorViews.java

```diff
@@ -181,6 +191,28 @@ private void lookupMatchInContentNode(LookupForResolvedViewsHelper utils, List<?
         });
     }
 
+    private int findFragmentStartOffsetInCurrentDocument(ContentNode contentNode, AbstractMatch match)
+    {
+        final int contentFragmentLength = contentNode.getAsXMLFragment().length();
+        final int matchStartOffset = match.getRange().getMinimumInteger();
+
+        if (contentFragmentLength < matchStartOffset) {
+            return matchStartOffset - contentFragmentLength;
+        }
+        return matchStartOffset;
+    }
+
+    private int findFragmentEndOffsetInCurrentDocument(ContentNode contentNode, AbstractMatch match, String currentDocumentContent)
+    {
+        final int contentFragmentLength = contentNode.getAsXMLFragment().length();
+        final int matchEndOffset = match.getRange().getMaximumInteger();
+
+        if (currentDocumentContent.length() > (matchEndOffset + contentFragmentLength)) {
+            return matchEndOffset + contentFragmentLength;
+        }
+        return matchEndOffset;
```

Comment:
After looking a bit closely I think there is an issue with the default return.
We should return `currentDocumentContent.length()` and not `matchEndOffset`

Reason: Returning `matchEndOffset` will make diff algorithm to run just on exact matched word